File: app/src/main/java/com/rankin/adam/cookingmaster/Model/ShoppingList.java
package com.rankin.adam.cookingmaster.Model;

import java.util.ArrayList;

/**
 * Created by Adam on 15-Jul-18.
 */

public class ShoppingList {

    private ArrayList<ShoppingListEntry> shoppingList;


    //TODO edit method to use amount and unit as well

    public ShoppingList() {
        this.shoppingList = new ArrayList<>();
    }

    public ArrayList<ShoppingListEntry> getShoppingList() {
        return shoppingList;
    }

    public void setShoppingList(ArrayList<ShoppingListEntry> ingredientsList) {
        this.shoppingList = ingredientsList;
    }

    public void addEntry(ShoppingListEntry shoppingListEntry){
        this.shoppingList.add(shoppingListEntry);
    }

    public void removeEntry(int position){
        this.shoppingList.remove(position);
    }

    public void clearShoppingList(){
        this.shoppingList.clear();
    }
}

File: app/src/main/java/com/rankin/adam/cookingmaster/model/ShoppingList.kt
package com.rankin.adam.cookingmaster.model

import java.util.ArrayList

/**
 * Created by Adam on 15-Jul-18.
 */
class ShoppingList {
    var shoppingList: ArrayList<ShoppingListEntry> = ArrayList()

    fun addEntry(shoppingListEntry: ShoppingListEntry) {
        shoppingList.add(shoppingListEntry)
    }

    fun removeEntry(position: Int) {
        shoppingList.removeAt(position)
    }

    fun clearShoppingList() {
        shoppingList.clear()
    }

    fun findIngredient(ingredient: Ingredient): ShoppingListEntry? {
        //TODO optimize
        for (element in shoppingList) {
            if (element.ingredient.name == ingredient.name) {
                return element
            }
        }
        return null
    }

    fun increaseAmount(position: Int, amount: Int) {
        shoppingList[position].addAmount(amount)
    }
}